libraries/chain/contracts/evt_contract.cpp

```diff
@@ -709,12 +709,9 @@ apply_evt_approvedelay(apply_context& context) {
 
         delay_def delay;
         tokendb.read_delay(adact.name, delay);
-        EVT_ASSERT(delay.status == delay_status::proposed, delay_status_exception, "Delay is not in 'proposed' status.");
+        EVT_ASSERT(delay.status == delay_status::proposed, delay_status_exception, "Delay is not in proper status.");
 
         auto signed_keys = delay.trx.get_signature_keys(adact.signatures, context.control.get_chain_id());
-        auto required_keys = context.control.get_delay_required_keys(delay.trx, signed_keys);
-        EVT_ASSERT(signed_keys == required_keys, delay_not_required_keys_exception, "Provided keys are not required in this delay transaction, provided keys: ${keys}", ("keys",signed_keys));
```

Comment:
Checks here is necessary, because all the keys signed must be required keys (aka. be involved in the authorization phase).
Otherwise he shouldn't sign his key.